k8s-version.txt

```diff
@@ -0,0 +1 @@
+1.16.3
```

Comment:
What is this file?

Reply:
I'm using `devctl k8s ensure` WIP feature by @tfussell to help validate that the k8s dependencies are of desired version https://github.com/giantswarm/devctl/pull/33

devctl needs to be released/published to be usable in CI builds, and for even more convenience devctl would have to be packaged in architect Docker image and a reusable job added in architect-orb.

It's all a workaround until k8s starts publishing v0.x.y go modules semantic import versioning compliant tags which may happen in 1.17 or 1.18 (see https://github.com/kubernetes/client-go/issues/670#issuecomment-552647733)

Reply:
Ack. Thanks for the explanation. Yes hard to know how much automation to build if these semver tags are coming.

Am I right in thinking helm 3 removes the import of `k8s.io/kubernetes`? Could that also improve the situation?

Reply:
Yes, helm 3 drops `k8s.io/kubernetes` dependency, it will reduce the noise in projects like ours which use helm as library and were getting `k8s.io/kubernetes` transitively. I've tried to ask helm maintainers to consider dropping it also for helm 2.15 and later for 2.16, but they refused, for backward compatibility reasons which I didn't quite understand (see https://github.com/helm/helm/pull/6609#issuecomment-544294599)

Reply:
Good that this improves with Helm 3 and Helm 2 seems to be in maintenance mode now. Thanks for checking on this.